feature-wallet-impl/src/main/java/jp/co/soramitsu/wallet/impl/presentation/balance/networkissues/NetworkIssuesViewModel.kt

```diff
@@ -0,0 +1,84 @@
+package jp.co.soramitsu.wallet.impl.presentation.balance.networkissues
+
+import androidx.lifecycle.asFlow
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import jp.co.soramitsu.account.api.domain.interfaces.AccountInteractor
+import jp.co.soramitsu.account.api.presentation.actions.AddAccountBottomSheet
+import jp.co.soramitsu.common.base.BaseViewModel
+import jp.co.soramitsu.common.compose.component.NetworkIssueItemState
+import jp.co.soramitsu.common.compose.component.NetworkIssueType
+import jp.co.soramitsu.common.mixin.api.NetworkStateMixin
+import jp.co.soramitsu.common.mixin.api.NetworkStateUi
+import jp.co.soramitsu.common.mixin.api.UpdatesMixin
+import jp.co.soramitsu.common.mixin.api.UpdatesProviderUi
+import jp.co.soramitsu.wallet.impl.domain.interfaces.WalletInteractor
+import jp.co.soramitsu.wallet.impl.presentation.WalletRouter
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+
+@HiltViewModel
+class NetworkIssuesViewModel @Inject constructor(
+    private val walletRouter: WalletRouter,
+    private val walletInteractor: WalletInteractor,
+    private val accountInteractor: AccountInteractor,
+    private val updatesMixin: UpdatesMixin,
+    private val networkStateMixin: NetworkStateMixin
+) : BaseViewModel(), UpdatesProviderUi by updatesMixin, NetworkStateUi by networkStateMixin {
+
+    val state = combine(
+        networkStateMixin.networkIssuesLiveData.asFlow(),
+        walletInteractor.assetsFlow().map {
+            it.filter { !it.hasAccount && !it.asset.markedNotNeed }.map {
+                NetworkIssueItemState(
+                    iconUrl = it.asset.token.configuration.chainIcon ?: it.asset.token.configuration.iconUrl,
+                    title = it.asset.token.configuration.chainName,
+                    type = NetworkIssueType.Account,
+                    chainId = it.asset.token.configuration.chainId,
+                    chainName = it.asset.token.configuration.chainName,
+                    assetId = it.asset.token.configuration.id
+                )
+            }
+        }
+    ) { networkIssues, assetsWoAccount ->
+        networkIssues.plus(assetsWoAccount)
+    }.map {
+        NetworkIssuesState(it)
+    }
+        .stateIn(
+            viewModelScope,
+            SharingStarted.Eagerly,
+            NetworkIssuesState(emptyList())
+        )
+
+    fun onIssueClicked(issue: NetworkIssueItemState) {
+        when (issue.type) {
+            NetworkIssueType.Node -> {
+                walletRouter.openNodes(issue.chainId)
+            }
+            NetworkIssueType.Network -> {
+                walletRouter.openNetworkUnavailable(issue.chainName)
+            }
+            NetworkIssueType.Account -> {
+                accountInteractor.selectedMetaAccountFlow().asLiveData().observeForever { meta ->
```

Comment:
Are you sure we need to subscribe to the meta account here? Especially using observeForever. I think it's enough here to just get the meta account once.